lib/IDE/Formatting.cpp

```diff
@@ -510,6 +510,7 @@ class FormatWalker : public SourceEntityWalker {
       bool operator==(const SourceLocIterator& rhs) {return It==rhs.It;}
       bool operator!=(const SourceLocIterator& rhs) {return It!=rhs.It;}
       SourceLoc operator*() {return It->getLoc();}
+      const SourceLoc operator*() const { return It->getLoc(); }
```

Comment:
SourceLoc is already a value type, so you could just make the existing one `const` instead.

Reply:
Yeah, I did consider that as well.  However, the important bit is the `const` on the `operator*` itself.  I figured that this was less intrusive.  I can definitely see if we can make the `operator*` const always as a follow up.

Reply:
@nkcsgexi, what do you think?

Reply:
This should be fine. Providing const overloads as well is a common pattern.